graveler/graveler_test.go

```diff
@@ -429,6 +429,142 @@ func TestGraveler_Commit(t *testing.T) {
+				RefManager:       &testutil.RefsFake{CommitID: expectedCommitID, Branch: &graveler.Branch{CommitID: expectedCommitID}, Commit: &graveler.Commit{TreeID: expectedTreeID}},
+			},
+			args: args{
+				ctx:          nil,
+				repositoryID: "repo",
+				branchID:     "branch",
+				committer:    "committer",
+				message:      "a message",
+				metadata:     nil,
+			},
+			want:        expectedCommitID,
+			expectedErr: graveler.ErrDirtyBranch,
+		},
+		{
+			name: "tree not found",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{Err: graveler.ErrNotFound},
+				StagingManager:   &testutil.StagingFake{ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{})},
+				RefManager:       &testutil.RefsFake{CommitID: expectedCommitID, Branch: &graveler.Branch{CommitID: expectedCommitID}, Commit: &graveler.Commit{TreeID: expectedTreeID}},
+			},
+			args: args{
+				ctx:          nil,
+				repositoryID: "repo",
+				branchID:     "branch",
+				committer:    "committer",
+				message:      "a message",
+				metadata:     nil,
+			},
+			want:        expectedCommitID,
+			expectedErr: graveler.ErrTreeNotFound,
+		},
+		{
+			name: "fail on add commit",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{TreeID: expectedTreeID, ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{})},
+				StagingManager:   &testutil.StagingFake{ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{})},
+				RefManager:       &testutil.RefsFake{CommitID: expectedCommitID, Branch: &graveler.Branch{CommitID: expectedCommitID}, CommitErr: graveler.ErrConflictFound, Commit: &graveler.Commit{TreeID: expectedTreeID}},
+			},
+			args: args{
+				ctx:          nil,
+				repositoryID: "repo",
+				branchID:     "branch",
+				committer:    "committer",
+				message:      "a message",
+				metadata:     nil,
+			},
+			want:        expectedCommitID,
+			expectedErr: graveler.ErrConflictFound,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			expectedCommitID := graveler.CommitID("expectedCommitId")
+			expectedTreeID := graveler.TreeID("expectedTreeID")
+			expectedBranchID := graveler.BranchID("expectedBranchID")
+			expectedRepositoryID := graveler.RepositoryID("expectedTreeID")
+			g := graveler.NewGraveler(tt.fields.CommittedManager, tt.fields.StagingManager, tt.fields.RefManager)
+			got, err := g.CommitExistingTree(context.Background(), expectedRepositoryID, expectedBranchID, expectedTreeID, tt.args.committer, tt.args.message, tt.args.metadata)
+			if !errors.Is(err, tt.expectedErr) {
+				t.Fatalf("unexpected err got = %v, wanted = %v", err, tt.expectedErr)
+			}
+			if err != nil {
+				return
+			}
+
+			// checking apply wasn't called
+			if diff := deep.Equal(tt.fields.CommittedManager.AppliedData, testutil.AppliedData{}); diff != nil {
```

Comment:
This is strange (but understandable perhaps): it tests that something did _not_ happen.

Reply:
Exactly! I want to make sure commit isn't done using Apply, otherwise it's a mess.

Reply:
I think I now better understand why this is "strange": it is a non-functional test.  Say I somehow managed to implement `Commit` using `Apply` and it was really elegant and efficient, and passed all the functional tests (i.e. a client would not be able to distinguish the results).  Then IIUC this test would fail, enforcing matters of our current good taste?

Why would we want to do this?  E.g. if we compacted some large uncommitted state into a "staging" tree (`git` style...) and wanted to commit _that_?  Dunno, anyway not sure about this part of the test.

OTOH, we've probably wasted enough time about this one line.

Reply:
Removed it.